app/controllers/application_controller.rb

```diff
@@ -22,9 +22,9 @@ def confirm_change(object, message = nil)
     flash[:message] = message
   end
 
-  def report_errors(object)
+  def report_errors(object, fallback_location)
```

Comment:
I might suggest a keyword argument, so that it's clear what passing a path means.